### PR TITLE
Ditch tomb2.exe dependency for TR2X

### PR DESCRIPTION
**Pull request by rr-**
#### Checklist

- [x] I have read the [coding conventions](https://github.com/LostArtefacts/TR1X/blob/master/CONTRIBUTING.md#coding-conventions)
- [x] I have added a changelog entry about what my pull request accomplishes, or it is an internal change

#### Description

Pulls all the variables to the local memory, initializing them where needed (most involved definitions concerned weapons and inventory ring items). Changes the build process to output a single .exe rather than a .dll + a process hollower, and no longer ships tomb2.exe.

### CODE REVIEW

**Comment by aredfan** [comment]
- [x] Crash when Lara is rolling
- [x] Crash when enabling fly cheat
- [x] Crash when `/play` 7/9
- [x] Crash when `/play` 16/17
- [x] Crash when using software renderer
- [x] Crash when Lara go near pickups
- [x] Crash when leaving the immediate start area of the Lara's Home (drydenj)
- [x] The UI background doesn't look quite right in the passport in SWR (drydenj)
- [x] the time within stats flashes a bit yellow on every update in SWR (drydenj)
- [x] the screen resolution appears to be 1919 * 1080 as that's the dimensions of the screenshots (drydenj)

**Comment by Richard-L** [comment]
Seem to be getting a crash in this room (could reproduce twice in a row):

https://github.com/user-attachments/assets/490398d3-f5e2-4cf9-8c6d-d5a0a4e6de77



Save: [savegame.zip](https://github.com/user-attachments/files/18223337/savegame.zip)

Can you reproduce?

I overwrote the new files into my existing TR2X test build folder, except `/cfg`. The `Tomb2.exe` is still present in this folder, so it's not a "fresh" setup. Should this at all be a problem?

I also notice:
- The main menu incorrectly starts on Lara's home and not the passport
- #2081 seems to be resolved in this build?

**Comment by Richard-L** [comment]
@aredfan simulpost 🙌 

Is it really just the roll? The first I tried I'm not sure that was what triggered it, but it's possible.

**Comment by rr-** [comment]
>The main menu incorrectly starts on Lara's home and not the passport

That's OG behavior, it'll likely become customizable in the gameflow.

>https://github.com/LostArtefacts/TRX/issues/2081 seems to be resolved in this build?

That's possible, yes.

**Comment by drydenj** [comment]
Yep roll crashes, and I get a crash when leaving the immediate start area of the Lara's Home
20241222_2210_crash

**Comment by rr-** [comment]
I appreciate everyone's reports. I've addressed the issues you mentioned; please let me know if you find any more bugs. Thank you.

CC @aredfan

**Comment by drydenj** [comment]
SW renderer indeed works again, however the image appears slightly out of proportion compared to previous working builds - the screen resolution appears to be 1919 * 1080 as that's the dimensions of the screenshots output from g5acf3b0. The UI background doesn't look quite right, and the time within stats flashes a bit yellow on every update.

load 0 7 1-177-g5acf3b0
load 0 7 1-166-g59bf9b3
stats 0 7 1-177-g5acf3b0
stats 0 7 1-166-g59bf9b3

**Comment by Richard-L** [comment]
Can't see anything wrong from a quick 30-40mins of testing.

**Comment by rr-** [comment]
@drydenj thanks – fixed.

File: README.md [hunk 1]
```diff
@@ -87,7 +87,8 @@ Please refer to the [detailed documentation](docs/tr1/).
 TR2X serves as a sequel to TR1X, currently focusing on the decompilation of Tomb Raider 2.
```
Comment (lahm86) [inline README.md]: Maybe we can merge this text with the paragraph below.

**Review by aredfan — APPROVED** [review]
LGTM 👍 

Hat tip to @drydenj for finding the remaining issues.

**Review by lahm86 — APPROVED** [review]
:clap:

Can confirm this also resolves #2081 for me. Not sure if we should capture that in the changelog as it was presumably a DLL side effect rather than a bug.

File: README.md (hunk 1 above)
Comment (rr-) [reply README.md]: Thanks – merged.

**Pull request merged**